pages/docs/couple-your-code/couple-your-code-defining-mesh-connectivity.md

```diff
@@ -95,3 +95,108 @@ void setMeshTriangles(int meshID, int size, int* vertices);
 void setMeshQuads(int meshID, int size, int* vertices);
 void setMeshTetrahedra(int meshID, int size, int* vertices);
 ```
+
+## Putting it all together
+
+Solvers may give you a range of information regarding vertices and faces.
+Some may give you unique ids for vertices, some only coordinates.
+In this section, we handle some common cases and how to implement them.
+
+### Solver provides IDs
+
+You solver gives each vertex or node a unique identifier.
+This identifier is also available when iterating over faces.
+
+In this case you can save a mapping from Solver ID to preCICE Vertex ID, after defining the vertices.
+When iterating over the faces, get the point identifier, map them to preCICE IDs and use those to define your connectivity.
+
+
+### Solver supports labels or tags
+
+You solver doesn't provide a unique identifier for each vertex or node.
+It does provide a functionality to attach some kind of label or tag to nodes.
```

Comment:
Do you have any particular example in mind for this case?

Reply:
Doesn't OpenFoam allow to do this?

Reply:
I was trying to understand which case OpenFOAM falls into, but I think it is the last one. I don't think every cell has a label, but every patch has faces associated with it. But I am not sure at all.